pmd-cli/src/main/java/net/sourceforge/pmd/cli/PmdCli.java

```diff
@@ -13,9 +13,13 @@ public final class PmdCli {
     private PmdCli() { }
 
     public static void main(String[] args) {
-        final int exitCode = new CommandLine(new PmdRootCommand())
-                .setCaseInsensitiveEnumValuesAllowed(true)
-                .execute(args);
-        System.exit(exitCode);
+        final CommandLine cli = new CommandLine(new PmdRootCommand())
+                .setCaseInsensitiveEnumValuesAllowed(true);
+        
+        // Don't show autocomplete subcommand in help by default
+        cli.getSubcommands().get("generate-completion")
+            .getCommandSpec().usageMessage().hidden(true);
```

Comment:
I was wondering if we should hide it always, or just under Windows where autocompletion is not available (except if using Cygwin…)

Reply:
On my machine the help message is the following with the command not hidden:
```
Usage: pmd [-hV] [COMMAND]
  -h, --help      Show this help message and exit.
  -V, --version   Print version information and exit.
Commands:
  check                The PMD standard source code analyzer
  cpd                  Copy/Paste Detector - find duplicate code
  designer             The PMD visual rule designer
  cpd-gui              GUI for the Copy/Paste Detector
                         Warning: May not support the full CPD feature set
  ast-dump             Experimental: dumps the AST of parsing source code
  generate-completion  Generate bash/zsh completion script for pmd.
Exit Codes:
  0   Successful analysis, no violations found
  1   An unexpected error occurred during execution
  2   Usage error, please refer to the command help
  4   Successful analysis, at least 1 violation found
```

I think having this line for `generate-completion` is very helpful, and it's also not clunky or confusing. Even on windows, people might be using a bash-like shell and would want to see this. Since it's clearly stated that it's for bash and zsh I don't think it's misleading.

Is the clunky text you're referring to somewhere else? Because otherwise I would say we definitely should not hide the command.

Reply:
Yes, I think we should _not_ hide this command. Just keep it simple. I'll double check, that the help doesn't look clunky...